spec/abci++/abci++_basic_concepts_002_draft.md

```diff
@@ -1,51 +1,269 @@
+As a general rule, an Application that detects an invalid vote extension SHOULD
+accept it in `ResponseVerifyVoteExtension` and ignore it in its own logic. Tendermint calls it when 
+a process receives a precommit message with a (possibly empty) vote extension.
+
+* [**FinalizeBlock:**](./abci++_methods_002_draft.md#finalizeblock) It delivers a decided block to the Application. The Application must execute the transactions in the block in order and update its state accordingly. Cryptographic commitments to the block and transaction results, via the corresponding
+parameters in `ResponseFinalizeBlock`, are included in the header of the next block. Tendermint calls it when a new block is decided.
+
+### Mempool methods
+
+For validating new transactions before they are included in the mempool.
+
+* [**CheckTx:**](./abci++_methods_002_draft.md#checktx) This method allows the Application to validate transactions against its current state, e.g., checking signatures and account balances. If a transaction passes the validation, then tendermint adds it to its local mempool, discarding it otherwise. Tendermint calls it when it receives a new transaction either coming from an external user or another node.
+
+### Info methods
+
+For initialization and queries from the user.
+
+* [**Info:**](./abci++_methods_002_draft.md#info) Used to sync Tendermint with the Application during a handshake that happens on startup.
+
+* [**Query:**](./abci++_methods_002_draft.md#query) Clients can use this method to query the Application for information about the application state.
+
+### State-sync methods
+
+State sync allows new nodes to rapidly bootstrap by discovering, fetching, and applying
+state machine snapshots instead of replaying historical blocks. For more details, see the
+[state sync section](../p2p/messages/state-sync.md).
+
+New nodes will discover and request snapshots from other nodes in the P2P network.
+A Tendermint node that receives a request for snapshots from a peer will call
+`ListSnapshots` on its Application to retrieve any local state snapshots. After receiving
+ snapshots from peers, the new node will offer each snapshot received from a peer
+to its local Application via the `OfferSnapshot` method.
 
-* For initialization and for queries from the user.
-* Handles the `Info` and `Query` calls.
+Snapshots may be quite large and are thus broken into smaller "chunks" that can be
+assembled into the whole snapshot. Once the Application accepts a snapshot and
+begins restoring it, Tendermint will fetch snapshot "chunks" from existing nodes.
+The node providing "chunks" will fetch them from its local Application using
+the `LoadSnapshotChunk` method.
 
-### **Snapshot** connection
+As the new node receives "chunks" it will apply them sequentially to the local
+application with `ApplySnapshotChunk`. When all chunks have been applied, the
+Application's `AppHash` is retrieved via an `Info` query. The `AppHash` is then
```

Comment:
I would clarify that this `AppHash` comparison is being done to ensure that the state sync was performed correctly. Consider adding a line that explains, such as, "To ensure that the sync proceeded correctly, Tendermint compares the local Application's `Apphash` to the `AppHash` stored on chain"